pennylane/ops/op_math/controlled.py

```diff
@@ -636,6 +628,10 @@ def has_decomposition(self):
         return False
 
     def decomposition(self):
+
+        if self.compute_decomposition is not Operator.compute_decomposition:
+            return self.compute_decomposition(*self.data, self.wires)
```

Comment:
Just like for `matrix`, this gets rid of a lot of duplicate code in `controlled_ops`.